DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/invocation_interfaces/src/streaming_pipelined_functor_model.cpp

```diff
@@ -0,0 +1,123 @@
+// oneAPI headers
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+#include "exception_handler.hpp"
+
+using ValueT = int;
+using MyUInt5 = ac_int<5, false>;
+
+struct FunctorStreamingPipelinedIP {
+  // Without the annotation, kernel argument will be inferred to be streaming
+  // kernel arguments if the kernel invocation interface is streaming, and
+  // vice-versa.             
+  ValueT* input;
+
+  // Annotate kernel argument with 'register_map' property 
+  // to explicitly specify it to be a register-mapped kernel argument.
+  sycl::ext::oneapi::experimental::annotated_arg<
+      ValueT *, decltype(sycl::ext::oneapi::experimental::properties{
+                    sycl::ext::intel::experimental::register_map})>                    
+      output;
+
+  // Kernel properties method to configure the kernel to be a kernel with 
+  // streaming pipelined invocation interface.
```

Comment:
suggested to add comment here for optional parameter so that user does not have to switch between readme and cpp file.
```suggestion
  // streaming pipelined invocation interface.
  // The property `sycl::ext::intel::experimental::pipelined` takes an optional template parameter that controls whether to pipeline the kernel. Valid parameters are:
  // -1: Pipeline the kernel, and automatically infer lowest possible II at target fMAX.
  // 0: Do not pipeline the kernel.
  // N (N> 0): Pipeline the kernel, and force the II of the kernel to be N.
  // If a parameter is not specified, the default behaviour of -1 will be inferred.
```